ibis/common/caching.py

```diff
@@ -47,56 +49,50 @@ def __init__(
         generate_name: Callable[[], str],
         key: Callable[[Any], Any],
     ) -> None:
-        self.cache = bidict()
-        # Somehow mypy needs a type hint here
-        self.refs: Counter = Counter()
         self.populate = populate
         self.lookup = lookup
         self.finalize = finalize
-        # Somehow mypy needs a type hint here
-        self.names: defaultdict = defaultdict(generate_name)
+        self.generate_name = generate_name
```

Comment:
Seems like there's only a single call site for this, can we inline the `generate_name` in `store` and remove the attribute altogether?

Reply:
Can also be done in a follow-up!

Reply:
Actually, nevermind, I'll submit a PR to clean some stuff up here that's independent of this PR!